preprocess/file_test.go

```diff
@@ -0,0 +1,47 @@
+package preprocess
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHasTextExtension(t *testing.T) {
+	require.False(t, hasTextExtension(""), "empty")
+	require.True(t, hasTextExtension("a.txt"), "known extension")
+	require.True(t, hasTextExtension("a.TXT"), "case insensitive")
+	require.False(t, hasTextExtension("a.jpg"), "unknown extension")
+	require.True(t, hasTextExtension("a/b/c.py"), "full path")
+}
+
+func TestHasShebang(t *testing.T) {
+	require.False(t, hasShebang(nil), "empty")
+
+	require.True(t, hasShebang([]byte("#! foo")), "shebang")
+
+	script := "\r\n  \t#!foo"
+	require.True(t, hasShebang([]byte(script)), "whitespace")
+
+	noscript := "    !$"
+	require.False(t, hasShebang([]byte(noscript)), "not shebang")
+
+	bomdata := append(bomSequences[1], []byte("    \t \n#!booyah")...)
+	require.True(t, hasShebang(bomdata), true)
```

Comment:
I would have said false.  Shebang should be at the absolute beginning, minus the BOM.  (Did I interpret that correctly?)